stackslib/src/chainstate/burn/db/sortdb.rs

```diff
@@ -1112,12 +1118,46 @@ pub trait SortitionHandle {
                 test_debug!("No snapshot at height {}", block_at_burn_height);
                 db_error::NotFoundError
             })?;
+        let top_sortition_id = sn.sortition_id;
+
+        let mut cache = DESCENDANCY_CACHE
+            .lock()
+            .expect("FATAL: lock poisoned in SortitionDB");
 
         while sn.block_height >= earliest_block_height {
+            match cache.get(&(sn.sortition_id, potential_ancestor.clone())) {
```

Comment:
I think you could get away with a single `let potential_ancestor = potential_ancestor.clone();` at the beginning at avoid all of these other clones.